plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1312 @@
+
+				if ejbValue, ok := subsystem["ejb3"]; ok {
+					ejb := ejbValue.(map[string]interface{})
+					tags := map[string]string{
+						"jboss_host":   host,
+						"jboss_server": serverName,
+						"name":         typeName,
+						"runtime_name": deploy.Result.RuntimeName,
+					}
+					h.processEJBAppStats(acc, ejb, tags)
+				}
+
+				if webValue, ok := subsystem["web"]; ok {
+					web := webValue.(map[string]interface{})
+					tags := map[string]string{
+						"jboss_host":   host,
+						"jboss_server": serverName,
+						"name":         typeName,
+						"runtime_name": deploy.Result.RuntimeName,
+					}
+					h.processWebAppStats(acc, web, tags)
+				}
+				//undertow is the new web sybsystem since wildfly 8
+				if webValue, ok := subsystem["undertow"]; ok {
+					web := webValue.(map[string]interface{})
+					tags := map[string]string{
+						"jboss_host":   host,
+						"jboss_server": serverName,
+						"name":         typeName,
+						"runtime_name": deploy.Result.RuntimeName,
+					}
+					h.processWebAppStats(acc, web, tags)
+				}
+
+			}
+			// This struct apply on WAR files
+			for typeName, value := range deploy.Result.Subsystem {
+				if value == nil {
+					log.Printf("D! JBoss Deployment SUBSYSTEM  value NULL")
+					continue
+				}
+				if typeName == "web" || typeName == "undertow" {
+					web := value.(map[string]interface{})
+					tags := map[string]string{
+						"jboss_host":   host,
+						"jboss_server": serverName,
+						"name":         deploy.Result.Name,
+						"runtime_name": deploy.Result.RuntimeName,
+					}
+					h.processWebAppStats(acc, web, tags)
+				} else {
+					log.Printf("W! JBoss Deployment WAR  from type %s", typeName)
+				}
+			}
+
+		}(deployment)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+// Flatten JSON hierarchy to produce field name and field value
+// Parameters:
+//    item: Item map to flatten
+//    fields: Map to store generated fields.
+//    id: Prefix for top level metric (empty string "")
+// Returns:
+//    void
+func (h *JBoss) flatten(item map[string]interface{}, fields map[string]interface{}, id string) {
+	if id != "" {
+		id = id + "_"
+	}
+	for k, i := range item {
+		switch i.(type) {
+		case int:
+			fields[id+k] = i.(float64)
+		case float64:
+			fields[id+k] = i.(float64)
+		case map[string]interface{}:
+			h.flatten(i.(map[string]interface{}), fields, id+k)
+		default:
+		}
+	}
+}
+
+func (h *JBoss) createRequestBody(optype int, address OrderedMap) (map[string]interface{}, error) {
+	bodyContent := make(map[string]interface{})
+
+	// Create bodyContent
+	switch optype {
+	case GetExecStat:
+		bodyContent = map[string]interface{}{
+			"operation":       "read-resource",
+			"attributes-only": "true",
+			"include-runtime": "true",
+			"address":         address,
+			"json.pretty":     1,
+		}
+	case GetHosts:
+		bodyContent = map[string]interface{}{
+			"operation":   "read-children-names",
+			"child-type":  "host",
+			"address":     address,
+			"json.pretty": 1,
+		}
```

Comment:
Please move those to an extra file and declare them as private (plugin-global) variables like
```suggestion
var execStatBody = map[string]interface{}{
	"operation":       "read-resource",
	"attributes-only": "true",
	"include-runtime": "true",
	"json.pretty":     1,
}

var hostsBody = map[string]interface{}{
	"operation":   "read-children-names",
	"child-type":  "host",
	"json.pretty": 1,
}
...
```

Then in the functions use those variables and add/set the `address`.